### ISSUE TITLE
hint tests failing

### ISSUE THREAD

**Issue by juhp** [ekmett/folds#12]
In the Stackage Nightly build:

```
> /tmp/stackage-build12/folds-0.7.1$ dist/build/hlint/hlint
src/Data/Fold.hs:123:64: Suggestion: Avoid lambda
Found:
  \ a x -> h x a
Why not:
  flip h

src/Data/Fold.hs:141:28: Suggestion: Avoid lambda
Found:
  \ a -> h a z
Why not:
  (`h` z)

src/Data/Fold.hs:221:73: Suggestion: Avoid lambda
Found:
  \ a -> Box (z a)
Why not:
  Box . z

src/Data/Fold.hs:224:72: Suggestion: Avoid lambda
Found:
  \ a -> Box (h z a)
Why not:
  Box . h z

src/Data/Fold/Internal.hs:69:3: Warning: Eta reduce
Found:
  foldl f z m0 = go m0
Why not:
  foldl f z = go

src/Data/Fold/Internal.hs:91:3: Warning: Eta reduce
Found:
  foldl f z m0 = go m0
Why not:
  foldl f z = go

src/Data/Fold/Internal.hs:95:3: Warning: Eta reduce
Found:
  foldl1 f m0 = go m0
Why not:
  foldl1 f = go

src/Data/Fold/Internal.hs:191:1: Suggestion: Use newtype instead of data
Found:
  data Box a = Box a
             deriving (Eq, Ord, Show, Read, Typeable, Data)
Why not:
  newtype Box a = Box a
                deriving (Eq, Ord, Show, Read, Typeable, Data)
Note: decreases laziness

src/Data/Fold/L'.hs:102:3: Warning: Redundant lambda
Found:
  (<$) b = \ _ -> pure b
Why not:
  (<$) b _ = pure b

src/Data/Fold/L'.hs:125:3: Warning: Redundant lambda
Found:
  (<*) m = \ _ -> m
Why not:
  (<*) m _ = m

src/Data/Fold/L'.hs:125:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L'.hs:160:3: Warning: Redundant lambda
Found:
  (<.) m = \ _ -> m
Why not:
  (<.) m _ = m

src/Data/Fold/L'.hs:160:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L'.hs:170:3: Warning: Redundant lambda
Found:
  (<@) m = \ _ -> m
Why not:
  (<@) m _ = m

src/Data/Fold/L'.hs:170:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L'.hs:198:3: Warning: Eta reduce
Found:
  cosieve (L' k0 h0 z0) as0 = go k0 h0 z0 as0
Why not:
  cosieve (L' k0 h0 z0) = go k0 h0 z0

src/Data/Fold/L.hs:109:3: Warning: Redundant lambda
Found:
  (<$) b = \ _ -> pure b
Why not:
  (<$) b _ = pure b

src/Data/Fold/L.hs:132:3: Warning: Redundant lambda
Found:
  (<*) m = \ _ -> m
Why not:
  (<*) m _ = m

src/Data/Fold/L.hs:132:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L.hs:167:3: Warning: Redundant lambda
Found:
  (<.) m = \ _ -> m
Why not:
  (<.) m _ = m

src/Data/Fold/L.hs:167:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L.hs:177:3: Warning: Redundant lambda
Found:
  (<@) m = \ _ -> m
Why not:
  (<@) m _ = m

src/Data/Fold/L.hs:177:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L.hs:204:3: Warning: Eta reduce
Found:
  cosieve (L k0 h0 z0) as0 = go k0 h0 z0 as0
Why not:
  cosieve (L k0 h0 z0) = go k0 h0 z0

src/Data/Fold/L1'.hs:51:43: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/L1'.hs:57:3: Warning: Redundant lambda
Found:
  (<.) m = \ _ -> m
Why not:
  (<.) m _ = m

src/Data/Fold/L1'.hs:57:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L1'.hs:63:42: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/L1'.hs:69:3: Warning: Redundant lambda
Found:
  (<*) m = \ _ -> m
Why not:
  (<*) m _ = m

src/Data/Fold/L1'.hs:69:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L1.hs:52:42: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/L1.hs:58:3: Warning: Redundant lambda
Found:
  (<.) m = \ _ -> m
Why not:
  (<.) m _ = m

src/Data/Fold/L1.hs:58:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/L1.hs:64:41: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/L1.hs:70:3: Warning: Redundant lambda
Found:
  (<*) m = \ _ -> m
Why not:
  (<*) m _ = m

src/Data/Fold/L1.hs:70:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/M.hs:47:14: Suggestion: Avoid lambda
Found:
  \ y -> k (m x y)
Why not:
  k . m x

src/Data/Fold/M.hs:68:15: Suggestion: Avoid lambda
Found:
  \ y -> k (m x y)
Why not:
  k . m x

src/Data/Fold/M.hs:71:15: Suggestion: Avoid lambda
Found:
  \ y -> k (m x y)
Why not:
  k . m x

src/Data/Fold/M.hs:116:3: Warning: Redundant lambda
Found:
  (<$) b = \ _ -> pure b
Why not:
  (<$) b _ = pure b

src/Data/Fold/M.hs:127:26: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/M.hs:137:3: Warning: Redundant lambda
Found:
  (<*) m = \ _ -> m
Why not:
  (<*) m _ = m

src/Data/Fold/M.hs:137:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/M.hs:173:3: Warning: Redundant lambda
Found:
  (<.) m = \ _ -> m
Why not:
  (<.) m _ = m

src/Data/Fold/M.hs:173:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/M.hs:183:3: Warning: Redundant lambda
Found:
  (<@) m = \ _ -> m
Why not:
  (<@) m _ = m

src/Data/Fold/M.hs:183:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/M1.hs:42:15: Suggestion: Avoid lambda
Found:
  \ y -> k (m x y)
Why not:
  k . m x

src/Data/Fold/M1.hs:59:28: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/M1.hs:65:3: Warning: Redundant lambda
Found:
  (<.) m = \ _ -> m
Why not:
  (<.) m _ = m

src/Data/Fold/M1.hs:65:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/M1.hs:71:27: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/M1.hs:77:3: Warning: Redundant lambda
Found:
  (<*) m = \ _ -> m
Why not:
  (<*) m _ = m

src/Data/Fold/M1.hs:77:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/R.hs:95:3: Warning: Redundant lambda
Found:
  (<$) b = \ _ -> pure b
Why not:
  (<$) b _ = pure b

src/Data/Fold/R.hs:136:3: Warning: Redundant lambda
Found:
  (<*) m = \ _ -> m
Why not:
  (<*) m _ = m

src/Data/Fold/R.hs:136:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/R.hs:153:3: Warning: Redundant lambda
Found:
  (<.) m = \ _ -> m
Why not:
  (<.) m _ = m

src/Data/Fold/R.hs:153:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/R.hs:163:3: Warning: Redundant lambda
Found:
  (<@) m = \ _ -> m
Why not:
  (<@) m _ = m

src/Data/Fold/R.hs:163:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/R.hs:188:3: Warning: Eta reduce
Found:
  cosieve (R k0 h0 z0) as0 = go k0 h0 z0 as0
Why not:
  cosieve (R k0 h0 z0) = go k0 h0 z0

src/Data/Fold/R1.hs:37:30: Suggestion: Avoid lambda
Found:
  \ c -> k (h a c)
Why not:
  k . h a

src/Data/Fold/R1.hs:52:42: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/R1.hs:58:3: Warning: Redundant lambda
Found:
  (<.) m = \ _ -> m
Why not:
  (<.) m _ = m

src/Data/Fold/R1.hs:58:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

src/Data/Fold/R1.hs:64:41: Suggestion: Use const
Found:
  \ _ -> ()
Why not:
  const ()

src/Data/Fold/R1.hs:70:3: Warning: Redundant lambda
Found:
  (<*) m = \ _ -> m
Why not:
  (<*) m _ = m

src/Data/Fold/R1.hs:70:12: Suggestion: Use const
Found:
  \ _ -> m
Why not:
  const m

69 hints
```

**Comment by phadej** [comment]
@juhp, could you rather disable tests in Stackage? I'm not sure about all of this hlint suggestions

**Comment by juhp** [comment]
Okay fair enough. :-)

**Comment by RyanGlScott** [comment]
The tests should be fixed in [`folds-0.7.2`](http://hackage.haskell.org/package/folds-0.7.2).

**Comment by RyanGlScott** [comment]
Oops, I forgot to include a small (but important) fix in `folds-0.7.2`. [`folds-0.7.3`](http://hackage.haskell.org/package/folds-0.7.3) should _really_ fix the `hlint` test suite.

**Comment by juhp** [comment]
Thanks
